apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/ApiDescriptionProcessor.java

```diff
@@ -443,46 +443,29 @@ private ApiDescriptionProcessor() {
 	 */
 	public static String serializeComponentXml(File location) {
 		if (location.exists()) {
-			ZipFile jarFile = null;
-			InputStream stream = null;
 			try {
 				String extension = IPath.fromOSString(location.getName()).getFileExtension();
 				if (extension != null && extension.equals("jar") && location.isFile()) { //$NON-NLS-1$
-					jarFile = new ZipFile(location, ZipFile.OPEN_READ);
-					ZipEntry manifestEntry = jarFile.getEntry(IApiCoreConstants.COMPONENT_XML_NAME);
-					if (manifestEntry != null) {
-						stream = jarFile.getInputStream(manifestEntry);
+					try (ZipFile jarFile = new ZipFile(location, ZipFile.OPEN_READ)) {
+						ZipEntry manifestEntry = jarFile.getEntry(IApiCoreConstants.COMPONENT_XML_NAME);
+						if (manifestEntry != null) {
+							try (InputStream stream = jarFile.getInputStream(manifestEntry)) {
+								return new String(Util.getInputStreamAsCharArray(stream, StandardCharsets.UTF_8));
+							}
```

Comment:
You could use `InputStream.readAllBytes()` here:
```suggestion
							byte[] allBytes = jarFile.getInputStream(manifestEntry).readAllBytes();
							return new String(allBytes, StandardCharsets.UTF_8);
```

Reply:
done